src/client-handler.ts

```diff
@@ -0,0 +1,46 @@
+import * as tmi from "tmi.js";
+import { IChatCommand } from "./chat-command.interface";
+import { GreetingCommand } from "./commands/greeting.model";
+
+const availableCommands: IChatCommand[] = [new GreetingCommand()];
+
+export function processClientMessage(target: string, sender: tmi.Userstate, msg: string) {
+    if (msg.indexOf("!") < 0) {
+        console.log(`* message dropped: no command - message: ${msg}, sender: ${sender.username}`);
+        return;
+    }
+
+    console.log(`* command received: no command - message: ${msg}, sender: ${sender.username}`);
+
+    const commandsWithReceiver = msg.match(/!\w+(\s*@[\w|\d]*)*/g);
+    if (!commandsWithReceiver || commandsWithReceiver.length <= 0) {
+        console.log(`* command dropped: no command - message: ${msg}, sender: ${sender.username}`);
+        return;
+    }
+
+    const firstCommandWithReceivers = commandsWithReceiver[0].trim().toLowerCase();
+    const matchingAvailableCommand = availableCommands.find(
+        (x) =>
+            (!Array.isArray(x.trigger) && firstCommandWithReceivers.indexOf(x.trigger.toLowerCase()) === 0) ||
+            (Array.isArray(x.trigger) &&
+                x.trigger.some((trigger) => firstCommandWithReceivers.indexOf(trigger.toLowerCase()) === 0)),
+    );
+
+    if (!matchingAvailableCommand) {
+        console.log("* dropping command: no handler found");
+        return;
+    }
+
+    if (
+        !matchingAvailableCommand.permittedUsers?.some((x) => x.toLowerCase() === sender.username?.toLowerCase()) &&
+        (matchingAvailableCommand.allowedForMods || matchingAvailableCommand.allowedForSubscriber) &&
+        (!(matchingAvailableCommand.allowedForMods && sender.mod) ||
+            !(matchingAvailableCommand.allowedForSubscriber && sender.subscriber))
+    ) {
+        console.log(`* dropping command: no permission ${sender.username}: ${msg}`);
+        return;
+    }
+
+    const recipients = firstCommandWithReceivers.match(/@[\w|\d]*/g);
```

Comment:
explanation comment for regexp maybe ?

Reply:
Oh, you want to explain briefly what my concern is with the RegEx? Have I understood that correctly?

Reply:
see the explanation for the other place with regexp. Only personal understanding problems... the code is totally fine.